libparsec/crates/platform_http_proxy/src/lib.rs

```diff
@@ -1,13 +1,27 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
 
+#[cfg(not(target_arch = "wasm32"))]
+mod native;
+#[cfg(target_arch = "wasm32")]
+mod wasm;
+#[cfg(target_os = "windows")]
+mod windows;
+
 use libparsec_types::prelude::*;
 
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::{HTTPS_PROXY, HTTP_PROXY};
+
 #[derive(Default, Debug, Clone)]
 pub struct ProxyConfig {
     #[cfg(not(target_arch = "wasm32"))]
     http_proxy: Option<reqwest::Proxy>,
     #[cfg(not(target_arch = "wasm32"))]
     https_proxy: Option<reqwest::Proxy>,
+    #[cfg(target_os = "windows")]
+    proxy_filter: Vec<String>,
+    #[cfg(target_os = "windows")]
+    proxy_script_url: String,
```

Comment:
Convert the value directly to `request::Proxy`

Reply:
`script_url` should be handled with the issue #4607, can you extract the change related to it ?

Reply:
The problem is that I have to let some `bytes` unused in the `register`, also the issue is about to support the `windows register`, not only the `proxy` server

Reply:
> Convert the values directly to `request::Proxy` (for `proxy_filter`)

According to the doc: https://docs.rs/reqwest/latest/reqwest/struct.Proxy.html#method.no_proxy

Reply:
Why do we have 2 proxies ? (`http` and `https` ?)

Reply:
> Why do we have 2 proxies ? (http and https ?)

I took example of curl that handle `[scheme]_proxy` [curl env var](https://curl.se/libcurl/c/libcurl-env.html), some other softwares also handle those type env variables (`wget`, `python.request` if I'm not mistaken because it surely use `libcurl` internally)

 
---

> > Convert the values directly to `request::Proxy` (for `proxy_filter`)
>
> According to the doc: https://docs.rs/reqwest/latest/reqwest/struct.Proxy.html#method.no_proxy

We currently don't handle `no-proxy` so we can ignore that for the moment (or provide support for the `NO_PROXY` env var)

---

> The problem is that I have to let some bytes unused in the register

We do you mean about `unused bytes in the register` ?

---

> the issue is about to support the windows register, not only the proxy server

Since we have another issue about supporting proxy pac, I would have expected that handling this case to be done on that issue.

Reply:
> We do you mean about `unused bytes in the register` ?

I mean this: https://github.com/Scille/parsec-cloud/pull/5749/files#diff-36d928dd01fae253e0c4f4b61a386f48de34e1ccdd266e3c0be6858bb5bf55f4R66-R73